Scalar.TestInfrastructure/Should/EnumerableShouldExtensions.cs

```diff
@@ -146,7 +146,7 @@ private static IEnumerable<T> ShouldMatch<T>(this IEnumerable<T> group, IEnumera
 
             if (shouldMatchInOrder)
             {
-                for (int i = 0; i < groupList.Count; ++i)
+                for (int i = 0; i < groupList.Count && i < expectedValuesList.Count; ++i)
```

Comment:
Shouldn't this be caught by the `if (groupList.Count != expectedValuesList.Count)` above?

Reply:
Oh we keep going after that. So this makes sense.

Reply:
Do we need to make this same change in the VFSforGit codebase?

Reply:
> Do we need to make this same change in the VFSforGit codebase?

Likely.

Reply:
The error I'm getting that is forcing this to happen is related to sparse-checkout, so it isn't needed for the `vfs-2.25.1` update. It would still be good to do.

Reply:
> Do we need to make this same change in the VFSforGit codebase?

This looks different in VFS for Git due to microsoft/vfsforgit@096226b72358a267208f89d067970983a3642865.